AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Models/AWSAuthCognitoSession.swift

```diff
@@ -77,6 +78,24 @@ public struct AWSAuthCognitoSession: AuthSession,
 
 }
 
+/// Internal Helpers for managing session tokens
+internal extension AWSAuthCognitoSession {
+    func areTokensExpiring(in seconds: TimeInterval? = nil) -> Bool {
+        
+        guard let tokens = try? cognitoTokensResult.get(),
+              let idTokenClaims = try? AWSAuthService().getTokenClaims(tokenString: tokens.idToken).get(),
+              let accessTokenClaims = try? AWSAuthService().getTokenClaims(tokenString: tokens.idToken).get(),
+              let idTokenExpiration = idTokenClaims["exp"]?.doubleValue,
+              let accessTokenExpiration = accessTokenClaims["exp"]?.doubleValue else {
+            return true
+        }
+        
+        // If the session expires < X minutes return it
+        return (Date(timeIntervalSince1970: idTokenExpiration).compare(Date(timeIntervalSinceNow: seconds ?? 0)) == .orderedDescending &&
```

Comment:
We used to compare against current time + 2 minute or something to expire early to avoid returning a session that will expire in next minute. Should we add that logic here?

Reply:
Its happening in the action ATM, so I kept this generic if we want to know something different in the future.. And the other thing that came to my mind was that action should drive that logic..